dem_handler/utils/spatial.py

```diff
@@ -509,3 +514,61 @@ def crop_datasets_to_bounds(
             # )
 
     return dem_array, dem_profile
+
+
+def check_dem_type_in_bounds(dem_type: str, bounds: BBox) -> bool:
+    """Check if the specified dem has data within the provided bounds. The provided dem_type is matched to either the
+    Copernicus Global 30m DEM or REMA DEM (currently implemented options). True is returned if the provided bounds
+    intersect with any tiles of the specified DEM.
+
+    Parameters
+    ----------
+    dem_type : str
+        dem type to check. Can be variations of REMA and COP_30. e.g. REMA_32, REMA_10,
+        cop30m, cop_30m, cop_glo30.
+    bounds : BBox
+        Bounds to check if data exists
+
+    Returns
+    -------
+    bool
+        True if the bounds intersects a tile, False otherwise.
+
+    Raises
+    -------
+    ValueError
+        If the provided dem_type cannot be matched to either the Copernicus 30m global DEM or the REMA dem.
+    """
+
+    if type(bounds) == BoundingBox:
```

Comment:
```suggestion
    if isinstance(bounds, BoundingBox):
```

`isinstance` is generally faster/more robust for checking a type, and is generally the pattern I've used elsewhere.

Reply:
If you could make this change in the `crop_datasets_to_bounds` function as well, that would be great. I can't suggest on it because it hasn't been changed in this PR